src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -6421,43 +6422,76 @@ def pivot_table(
-                values_label_to_identifier_pairs_list,
-                aggfunc,
-                len(pivot_snowflake_quoted_identifiers) > 0,
-                isinstance(values, list)
-                and (not margins or len(values) > 1)
-                and include_aggr_func_in_label,
-                sort,
+            multiple_agg_funcs_single_values = (
+                isinstance(aggfunc, list) and len(aggfunc) > 1
+            ) and not isinstance(values, list)
+            include_aggr_func_in_label = (
+                len(groupby_snowflake_quoted_identifiers) != 0
+                or multiple_agg_funcs_single_values
+            )
+            pivot_aggr_groupings = list(
+                generate_single_pivot_labels(
+                    values_label_to_identifier_pairs_list,
+                    aggfunc,
+                    len(pivot_snowflake_quoted_identifiers) > 0,
+                    isinstance(values, list)
+                    and (not margins or len(values) > 1)
+                    and include_aggr_func_in_label,
+                    sort,
+                )
             )
-        )
+        else:
+            # If there are no values, we simply return an empty DataFrame with no columns
+            # whose index is the result of grouping by the index columns. We pass
+            # pivot_aggr_groupings = None to signify that case in the `pivot_helper`
+            # function.
+            pivot_aggr_groupings = None
 
         # When aggfunc is not a list, we should sort the outer level of pandas labels.
-        pivotted_frame = pivot_helper(
-            self._modin_frame,
-            pivot_aggr_groupings,
-            not dropna,
-            not isinstance(aggfunc, list),
-            columns,
-            groupby_snowflake_quoted_identifiers,
-            pivot_snowflake_quoted_identifiers,
-            (isinstance(aggfunc, list) and len(aggfunc) > 1),
-            (isinstance(values, list) and len(values) > 1),
-            index,
-        )
+        try:
+            pivotted_frame = pivot_helper(
+                self._modin_frame,
+                pivot_aggr_groupings,
+                not dropna,
+                not isinstance(aggfunc, list),
+                columns,
+                groupby_snowflake_quoted_identifiers,
+                pivot_snowflake_quoted_identifiers,
+                (isinstance(aggfunc, list) and len(aggfunc) > 1),
+                (isinstance(values, list) and len(values) > 1),
+                index,
+            )
+        except SnowparkSQLException as e:
+            # Error Code 1146 corresponds to the Snowflake Exception when
+            # a dynamic pivot is called and there are no pivot values and no
+            # groupby columns specified. If we hit this error, that means that
+            # we have attempted a pivot on an empty DataFrame, so we catch
+            # the exception and return an empty DataFrame.
+            if e.sql_error_code == 1146:
```

Comment:
For this behavior, we do rely on the fact that a describing query is triggered for the underneath pivot in ordered_dataframe, right? can you add a comment here to be clear

Reply:
Sure - (we rely on the fact that we eagerly grab the pivot values in dynamic pivot, and so if there are no pivot values and no groupby columns we throw an exception eagerly from the server side) - will add a comment to that effect!

Reply:
where did we add this? and please also add the fact that we are relying on the fact that we eagerly trigger the describing query to grab pivot result column, if that is not the case, no eager grabbing of pivot value will be triggerred

Reply:
> where did we add this?
The feature I described above is a part of the dynamic pivot implementation.

If you mean where is the comment, it should be added right above here?

The error is raised before we even attempt to materialize anything client side, as the compilation of a dynamic pivot query involves eagerly materializing the pivot values, and erroring out if no groupby columns are specified and no pivot values are found.

Reply:
sorry, i was asking where did we put the comment,  i don't think i see the comment before. 

I think my point is that our this does require the fact that we eagerly triggered a metadata query (describe) for pivot, which will trigger a compilation on the server side, so people can be careful when they are replying on catching this error. However, i don't think we will be able to get rid of the metadata query in any short term, which should be fine.

Reply:
I see - so should I update the comment or is it good as is?

Reply:
i think would be good to add that in comment so that future developer is aware of this.